File: dev-docs/snippets/documents/list.ts
import type { JsBaoClient } from "js-bao-wss-client";

// List documents accessible to the current user. Deprecated in favor of
// `client.me.ownedDocuments()` / `client.me.sharedDocuments()`.
export async function list(client: JsBaoClient) {
  // #region example
  const docs = await client.documents.list({ tag: "planning", limit: 50 });
  // #endregion example
  return docs;
}

File: dev-docs/snippets/documents/list.swift
import JsBaoClient

// List documents accessible to the current user as a typed `[DocumentInfo]`.
// (Deprecated — prefer `client.me.ownedDocuments()` / `sharedDocuments()`.)
func list(client: JsBaoClient) async throws {
  // #region example
  let items = try await client.documents.list(
    options: PaginationOptions(limit: 50)
  )
  let firstTitle = items.first?.title
  // #endregion example
  _ = firstTitle
}
